speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -186,6 +257,57 @@ static void SaveSettings(void* /* dummy */)
     GfuiScreenActivate(s_nextHandle);
 }
 
+/// @brief Inializes the menu setting from the ResearcherMenu.xml file
+void initializeSettings(void* p_param) {
+    // Retrieve all setting variables from the xml file
+    bool checkboxTaskLaneKeeping = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTaskLaneKeeping", "checked", NULL), false);
+    bool checkboxTaskSpeedControl = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTaskSpeedControl", "checked", NULL), false);
+    bool CheckboxIndicatorAuditory = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxIndicatorAuditory", "checked", NULL), false);
+    bool checkboxIndicatorVisual = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxIndicatorVisual", "checked", NULL), false);
+    bool checkboxTypeNoSignals = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeNoSignals", "checked", NULL), false);
+    bool CheckboxTypeOnlySignals = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeOnlySignals", "checked", NULL), false);
+    bool checkboxTypeSharedControl = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeSharedControl", "checked", NULL), false);
+    bool checkboxPControlGas = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxPControlGas", "checked", NULL), false);
+    bool checkboxInterventionToggle = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxPControlInterventionToggle", "checked", NULL), false);
+    bool checkboxPControlSteering = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxPControlSteering", "checked", NULL), false);
+
+    // Set the max time setting from the xml file
+    m_maxTime = atoi(GfParmGetStr(p_param, "dynamic controls/MaxTimeEdit", "default value", NULL));
+    
+
+    // Set the Task settings from the xml file
+    if (!checkboxTaskLaneKeeping) {
+        m_task = checkboxTaskSpeedControl ? TASK_SPEED_CONTROL : TASK_NO_TASK;
+    }
+    else {
+        m_task = TASK_LANE_KEEPING;
+    }
+
+    // Set the indicator settings from the xml file
+    m_indicators.Auditory = CheckboxIndicatorAuditory;
+    m_indicators.Visual = checkboxIndicatorVisual;
+
+    // Set the indicator settings from the xml file
+    m_pControl.ControlGas = checkboxPControlGas;
+    m_pControl.ControlInterventionToggle = checkboxInterventionToggle;
```

Comment:
Keep order consistent
```suggestion
    m_pControl.ControlInterventionToggle = checkboxInterventionToggle;
    m_pControl.ControlGas = checkboxPControlGas;
```